packages/react-composites/src/composites/ChatComposite/ChatScreen.tsx

```diff
@@ -216,6 +226,141 @@ export const ChatScreen = (props: ChatScreenProps): JSX.Element => {
 
   const userId = toFlatCommunicationIdentifier(adapter.getState().userId);
 
+  /* @conditional-compile-remove(rich-text-editor-image-upload) */
+  const fetchBlobData = useCallback(
+    async (
+      resource: string | URL | Request,
+      options: { timeout?: number; headers?: Headers; abortController: AbortController }
+    ): Promise<Response> => {
+      // default timeout is 30 seconds
+      const { timeout = 30000, abortController } = options;
+
+      const id = setTimeout(() => {
+        abortController.abort();
+      }, timeout);
+
+      const response = await fetch(resource, {
+        ...options,
+        signal: abortController.signal
+      });
+      clearTimeout(id);
+      return response;
+    },
+    []
+  );
+
+  /* @conditional-compile-remove(rich-text-editor-image-upload) */
+  const getInlineImageData = useCallback(
+    async (image: string): Promise<Blob | undefined> => {
+      const blobImage: Blob | undefined = undefined;
+      if (image.startsWith('blob') || image.startsWith('http')) {
+        const res = await fetchBlobData(image, { abortController: new AbortController() });
+        const blobImage = await res.blob();
+        return blobImage;
+      }
+      return blobImage;
+    },
+    [fetchBlobData]
+  );
+
+  /* @conditional-compile-remove(rich-text-editor-image-upload) */
+  const inlineImageUploadHandler = useCallback(
+    async (uploadTasks: AttachmentUpload[]): Promise<void> => {
+      for (const task of uploadTasks) {
+        const uploadTask = task as AttachmentUploadTask;
+        const image: Blob | undefined = uploadTask.image;
+        if (!image) {
+          uploadTask.notifyUploadFailed(`Image data for "${task.metadata?.name}" is not provided.`);
+          continue;
+        }
+        if (image && image.size > MAX_INLINE_IMAGE_UPLOAD_SIZE_MB * 1024 * 1024) {
+          uploadTask.notifyUploadFailed(
+            `"${task.metadata?.name}" is too big. Select a file under ${MAX_INLINE_IMAGE_UPLOAD_SIZE_MB}MB.`
+          );
+          continue;
+        }
+
+        const SUPPORTED_FILES: Array<string> = ['jpg', 'jpeg', 'png', 'gif', 'heic', 'webp'];
```

Comment:
Are these all supported by the backend? Is there a way we can configure this to be queried rather than hardcoded?

Reply:
> Are these all supported by the backend? Is there a way we can configure this to be queried rather than hardcoded?

Yes, this is the list that supported by the backend. By queried, are you thinking of whether there's an API that dynamically fetch this list? Or we set this in some config file instead of having it in the ChatScreen?